jdaviz/app.py

```diff
@@ -342,6 +343,102 @@ def get_data_from_viewer(self, viewer_reference, data_label=None, cls=None,
 
         return data
 
+    def get_subsets_from_viewer(self, viewer_reference, data_label=None):
+        """
+        Returns the subsets of a specified viewer converted to astropy regions
+        objects.
+
+        It should be noted that the subset translation machinery lives in the
+        glue-astronomy repository. Currently, the machinery only works on 2D
+        data for cases like range selection. For e.g. a profile viewer that is
+        ostensibly just a view into a 3D data set, it is necessary to first
+        reduce the dimensions of the data, then retrieve the subset information
+        as a regions object. This means that the returned y extents in the
+        region are not strictly representative of the subset range in y.
+
+        Parameters
+        ----------
+        viewer_reference : str
+            The reference to the viewer defined with the ``reference`` key
+            in the yaml configuration file.
```

Comment:
For a user-level API this is pretty opaque - the user needs to be able to get these regions without looking at a configuration file. I realize this is used elsewhere though, so I think this is more of a separate issue and I think it neednt hold up this particular PR.

Reply:
(see #175)